### PR TITLE
Bug 1728877: restrict SNI config to avoid overriding internal ELB

### PR DESCRIPTION
**Pull request by deads2k**
The internal LB is used by kubelets which cannot update their trust. This means that if the internal LB certificate is changed, the cluster loses connectivity. We auto-rotate the serving cert, but not the corresponding trust for it.

backport of https://github.com/openshift/origin/pull/23297  this is not a clean pick.  Please re-review

/assign @sttts

### CODE REVIEW

File: pkg/admission/customresourcevalidation/apiserver/validate_apiserver_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
+package apiserver
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
```
Comment (sttts) [inline pkg/admission/customresourcevalidation/apiserver/validate_apiserver_test.go]: nit: order

**Comment by sttts** [comment]
lgtm

**Comment by sttts** [comment]
/hold

APIServer validation is broken.

File: pkg/admission/customresourcevalidation/apiserver/validation_wrapper.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+// Register registers a plugin
+func Register(plugins *kadmission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (kadmission.Interface, error) {
+		return NewValidateAPIServer()
+	})
+}
+
+type validateCustomResourceWithClient struct {
+	kadmission.Interface
```
Comment (sttts) [inline pkg/admission/customresourcevalidation/apiserver/validation_wrapper.go]: this is still wrong

**Comment by sttts** [comment]
/lgtm

Tide is unhappy.

**Comment by deads2k** [comment]
/hold cancel

fixed

**Comment by sttts** [comment]
@eparis this need the cherry-pick-approved label to merge.

**Comment by derekwaynecarr** [comment]
i think the automated unit test provides sufficient coverage.

approving for this week.

**Pull request merged**